DuckDB.NET.Data/DuckDBQueryResult.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+
+namespace DuckDB.NET.Data;
+
+internal sealed class DuckDBQueryResult : IDisposable
```

Comment:
What is the reason for adding this class?

Reply:
Why not make `DuckDBResult` itself implement `IDisposable`?

Reply:
I think it's ok for the native structure to implement `IDisposable` because other native components' counterparts already implement from `SafeHandle` and hence implement `IDisposable`.